quote.go

```diff
@@ -0,0 +1,356 @@
+}
+
+// Data used to generate a new [Price](https://stripe.com/docs/api/prices) object inline. One of `price` or `price_data` is required.
+type QuoteLineItemPriceDataParams struct {
+	Currency          *string                                `form:"currency"`
+	Product           *string                                `form:"product"`
+	Recurring         *QuoteLineItemPriceDataRecurringParams `form:"recurring"`
+	TaxBehavior       *string                                `form:"tax_behavior"`
+	UnitAmount        *int64                                 `form:"unit_amount"`
+	UnitAmountDecimal *float64                               `form:"unit_amount_decimal,high_precision"`
+}
+
+// A list of line items the customer is being quoted for. Each line item includes information about the product, the quantity, and the resulting cost.
+type QuoteLineItemParams struct {
+	ID        *string                       `form:"id"`
+	Price     *string                       `form:"price"`
+	PriceData *QuoteLineItemPriceDataParams `form:"price_data"`
+	Quantity  *int64                        `form:"quantity"`
+	TaxRates  []*string                     `form:"tax_rates"`
+}
+
+// When creating a subscription or subscription schedule, the specified configuration data will be used. There must be at least one line item with a recurring price for a subscription or subscription schedule to be created. A subscription schedule is created if `subscription_data[effective_date]` is present and in the future, otherwise a subscription is created.
+type QuoteSubscriptionDataParams struct {
+	EffectiveDate                 *int64 `form:"effective_date"`
+	EffectiveDateCurrentPeriodEnd *bool  `form:"-"` // See custom AppendTo
+	TrialPeriodDays               *int64 `form:"trial_period_days"`
+}
+
+// AppendTo implements custom encoding logic for QuoteSubscriptionDataParams.
+func (q *QuoteSubscriptionDataParams) AppendTo(body *form.Values, keyParts []string) {
+	if BoolValue(q.EffectiveDateCurrentPeriodEnd) {
+		body.Add(form.FormatKey(append(keyParts, "effective_date")), "current_period_end")
+	}
+}
+
+// The data with which to automatically create a Transfer for each of the invoices.
+type QuoteTransferDataParams struct {
+	Amount        *int64   `form:"amount"`
+	AmountPercent *float64 `form:"amount_percent"`
+	Destination   *string  `form:"destination"`
+}
+
+// Clone an existing quote. The new quote will be created in `status=draft`. When using this parameter, you cannot specify any other parameters except for `expires_at`.
+type QuoteFromQuoteParams struct {
+	IsRevision *bool   `form:"is_revision"`
+	Quote      *string `form:"quote"`
+}
+
+// Returns a list of your quotes.
+type QuoteListParams struct {
+	ListParams `form:"*"`
+	Customer   *string `form:"customer"`
+	Status     *string `form:"status"`
+}
+
+// Cancels the quote.
+type QuoteCancelParams struct {
+	Params `form:"*"`
+}
+
+// Finalizes the quote.
+type QuoteFinalizeQuoteParams struct {
```

Comment:
For Invoices, we shortened these to `InvoiceFinalizeParams`. Do we want to do the same here?

Reply:
Spoke offline, this is OK to leave as is. While not consistent in this case, we don't want to have to special case this every time.